packages/components/src/mobile/bottom-sheet/cell.native.js

```diff
@@ -57,6 +57,18 @@ class BottomSheetCell extends Component {
 		this.setState( { isScreenReaderEnabled } );
 	}
 
+	typeToKeyboardType( type, step ) {
+		let keyboardType = `default`;
+		if ( type === `number` ) {
+			if ( step && Math.abs( step ) < 1 ) {
+				keyboardType = `decimal-pad`;
+			} else {
+				keyboardType = `number-pad`;
+			}
+		}
+		return keyboardType;
+	}
```

Comment:
I was thinking first to have this in the TextControl, but if it's something that could be used in other instances of Cell I think it's fine to be here.

I'm wondering, what is `step` used for?
The settings sends a `step="1"` and it seems to not be used by the Web `TextControl` or html `textarea` 🤔

Reply:
From what I see from the web the step prop is passed directly to HTML input element. 
On the web step is used like this: https://www.w3schools.com/tags/att_input_step.asp

And from what I see TextControl on native uses Cell directly, how do you see it I can use it on the TextControl level of native?

Reply:
> And from what I see TextControl on native uses Cell directly, how do you see it I can use it on the TextControl level of native?

Any extra prop we pass to Cell, will be added to the Cell's internal `TextInput` component via the `valueProps` here: https://github.com/WordPress/gutenberg/blob/bff7d165982da86fb938c151e0c08abd8a31af3c/packages/components/src/mobile/bottom-sheet/cell.native.js#L174

So if we pass `keyboardType` to Cell in `TextControl` it should work. Since Cell is more generic and TextControl is meant to be used as the text input component, maybe it would make sense. But I'm fine with keeping it in Cell how it is now too. Give a thought and you can decide :)

Thanks for the explanation, I didn't see the prop on the `textarea` docs but I see the arrows up and down in the Web settings UI, so that makes sense 👍